tests/run-make/rust-lld-custom-target/custom-target.json

```diff
@@ -0,0 +1,54 @@
+{
+  "arch": "x86_64",
+  "cpu": "x86-64",
+  "crt-static-respected": true,
+  "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+  "dynamic-linking": true,
+  "env": "gnu",
+  "has-rpath": true,
+  "has-thread-local": true,
+  "link-self-contained": {
```

Comment:
Currently, `link-self-contained` is serialized as `crt-objects-fallback`, and this nesting level is made to allow fixing that in the future: by incorporating the musl/mingw inference strategies, which don't map to a full/empty list of enabled components. 

If we want to have all self-contained linking things in the same place with the correct names, then yes it seems interesting to me, in order to not have another case of backwards-compatible parsing schemes when we define a final stable target specs json schema in the future. 

Otherwise if not, and we're fine with splitting into 2 keys and removing the nesting, then a better name would probably look something like `"link-self-contained-components": ["linker"]` instead.

The nesting here prepares such a possibility of removing `"crt-objects-fallback" = "true"` and "false" by replacing them with full/empty components. Then, having inference strategies as e.g.

```json
"link-self-contained": {
  "infer": "musl"
}
```

grouped under the same key as target specs enabling a given component

```json
"link-self-contained": {
  "components": ["linker"]
}

Reply:
Ok.